modules/map/src/mapapi.c

```diff
@@ -1164,7 +1164,7 @@ MAP_EXTERNCALL void map_set_gravity(MAP_ParameterType_t* p_type, const double gr
 
 MAP_EXTERNCALL void map_set_input_text(MAP_InitInputType_t* init_type, const char* input_txt_line)
 {
-  strncpy(init_type->library_input_str, input_txt_line, 254);
+  strncpy_s(init_type->library_input_str, sizeof init_type->library_input_str, input_txt_line, (sizeof init_type->library_input_str)-1);
```

Comment:
I propose the following, which doesn't doesn't use the `str*` functions and guarantees that the string will be null terminated. Which is essentially what `strncpy` should do.

```suggestion
  int i = 0;
  for (i = 0; i < (sizeof init_type->library_input_str); ++i)
  {
    init_type->library_input_str[i] = input_txt_line[i];
    if (input_txt_line[i] == 0) break;
  }
  if (init_type->library_input_str[i] != 0)
  {
    init_type->library_input_str[i] = 0;
  }
```